game/nintendo_mii/gen2_wiiu_3ds_miitomo_mii.ksy

```diff
@@ -0,0 +1,188 @@
+meta:
+  id: gen2_wiiu_3ds_miitomo_mii
+  file-extension:
+    - cfsd
+    - ffsd
+  endian: le
+seq:
+  - id: unknown_1
+    type: u1
+    doc: Always 3?
+  - id: character_set
+    type: b2
+    enum: character_set
+  - id: region_lock
+    type: b2
+    enum: region_lock
+  - id: profanity_flag
+    type: b1
+  - id: copying
+    type: b1
+  - id: unknown_2
+    type: b2
+  - id: mii_position_slot_index
+    type: b4
+  - id: mii_position_page_index
+    type: b4
+  - id: version
+    type: b4
+  - id: unknown_3
+    type: b4
+  - id: system_id
+    type: u1
+    repeat: expr
+    repeat-expr: 8
+  - id: avatar_id
+    type: u1
+    repeat: expr
+    repeat-expr: 4
+  - id: client_id
+    type: u1
+    repeat: expr
+    repeat-expr: 6
+  - id: padding
+    type: u2
+  - id: data_1
+    type: u2
+  - id: mii_name
+    type: str
+    size: 20
+    encoding: utf-16
+  - id: body_height
+    type: u1
+  - id: body_weight
+    type: u1
+  - id: face_color
+    type: b3
+  - id: face_type
+    type: b4
+  - id: mingle
+    type: b1
+  - id: face_makeup
+    type: b4
+  - id: face_wrinkles
+    type: b4
+  - id: hair_type
+    type: u1
+  - id: unknown_5
+    type: b4
+  - id: hair_flip
+    type: b1
+  - id: hair_color
+    type: b3
+  - id: eye
+    type: u4
+  - id: eyebrow
+    type: u4
+  - id: nose
+    type: u2
+  - id: mouth
+    type: u2
+  - id: mouth2
+    type: u2
+  - id: beard
+    type: u2
+  - id: glasses
+    type: u2
+  - id: mole
+    type: u2
+  - id: creator_name
+    type: str
+    size: 20
+    encoding: utf-16
+  - id: padding2
+    type: u2
+  - id: checksum
+    type: u2
+enums:
+  character_set:
+    0: jpn_usa_eur
+    1: chn
+    2: kor
+    3: twn
+  region_lock:
+    0: no_lock
+    1: jpn
+    2: usa
```

Comment:
We have "bit-sized types" that may be useful for more straightforward specifying of this.

Reply:
This is counter to [the advice given in the previous thread](https://github.com/kaitai-io/kaitai_struct_formats/pull/356#discussion_r526391246). Are you sure that's better?

Reply:
@KOLANICH 

> We have "bit-sized types" that may be useful for more straightforward specifying of this.

Um, you're talking about the `instances` a few lines below, not this enum `region_lock`, right?

Sure, after seeing the `instances` heavily relying on manual bit handling, I also consider it necessary to rewrite it using the [bit-sized integers](https://doc.kaitai.io/user_guide.html#_bit_sized_integers).

Reply:
Yes, seems like attached to a wrong line.